### ISSUE TITLE
Directory "database-incognito" automatically created in ~/

### ISSUE THREAD

**Issue by bheadmaster** [qutebrowser/qutebrowser#4320]
Having 'content.private_browsing' turned on, qutebrowser often creates a directory named 'databases-incognito' in the home directory. I can't seem to find a way to turn this off.

I have also found a similar bug with QupZilla from January:
https://github.com/QupZilla/qupzilla/issues/2547

***

qutebrowser v1.5.0
Git commit:
Backend: QtWebEngine (Chromium 65.0.3325.230)
 
CPython: 3.7.0
Qt: 5.11.2
PyQt: 5.11.3
 
sip: 4.19.13
colorama: no
pypeg2: 2.15
jinja2: 2.10
pygments: 2.2.0
yaml: 3.13
cssutils: no
attr: 18.2.0
PyQt5.QtWebEngineWidgets: yes
PyQt5.QtWebKitWidgets: no
pdf.js: no
sqlite: 3.25.2
QtNetwork SSL: OpenSSL 1.1.1  11 Sep 2018
 
Style: QFusionStyle
Platform: Linux-4.18.12-arch1-1-ARCH-x86_64-with-arch, 64bit
Linux distribution: Arch Linux (arch)
Frozen: False
Imported from /usr/lib/python3.7/site-packages/qutebrowser
Using Python from /usr/bin/python3
Qt library executable path: /usr/lib/qt/libexec, data path: /usr/share/qt
 
Paths:
cache: /home/lain/.cache/qutebrowser
config: /home/lain/.config/qutebrowser
data: /home/lain/.local/share/qutebrowser
runtime: /run/user/1000/qutebrowser
system data: /usr/share/qutebrowser
 
Uptime: 0:48:52

**Comment by The-Compiler** [comment]
As per the QupZilla issue, this is a bug in QtWebEngine: https://bugreports.qt.io/browse/QTBUG-62957

**Comment by marvinborner** [comment]
Yep. I solved it with this non-portable and probably extremely dumb method of replacing the `/usr/bin/qutebrowser` script with the following:

```python3
#!/usr/bin/python3
__requires__ = 'qutebrowser==1.12.0'
import re
import sys
import shutil
import os
from pkg_resources import load_entry_point

def remove(dirname):
    try:
        shutil.rmtree(dirname)
    except OSError as e:
        print("Error: %s - %s." % (e.filename, e.strerror))

if __name__ == '__main__':
    sys.argv[0] = re.sub(r'(-script\.pyw?|\.exe)?$', '', sys.argv[0])
    reason = load_entry_point('qutebrowser==1.12.0', 'gui_scripts', 'qutebrowser')()

    # Remove directories
    username = os.getlogin();
    remove('/home/%s/databases-incognito/' % username);
    remove('/home/%s/VideoDecodeStats/' % username);
    remove('/home/%s/shared_proto_db/' % username);

    # Exit
    sys.exit(reason)
```

**Comment by The-Compiler** [comment]
@marvinborner FWIW to make this a bit more portable, you can use `os.path.expanduser('~/...')` to get the paths.

I thought about adding a similar workaround inside qutebrowser, but I don't feel comfortable with running the equivalent of `rm -r` on folders in the user's home directory which aren't clearly attributable to qutebrowser. I think the risks of adding such a workaround are bigger than the benefits it would bring.

Thankfully, this seems to be fixed now in QtWebEngine for Qt 5.15.1: [Return valid path in Profile::GetPath() for incognito profiles (Iec6e2455) · Gerrit Code Review](https://codereview.qt-project.org/c/qt/qtwebengine/+/305361)

**Comment by The-Compiler** [comment]
FWIW on Archlinux, this is [fixed](https://git.archlinux.org/svntogit/packages.git/commit/trunk?h=packages/qt5-webengine&id=fc7cbcf3af5fe895e7f19a2b8ce6333c150c362c) in qt5-webengine 5.15.0-3 which backports the upstream fix.
